src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -110,7 +110,7 @@ public Pipe(PipeOptions options)
 
             // If we're using the default pool then mark it as null since we're just going to use the
             // array pool under the covers
-            _pool = options.Pool == MemoryPool<byte>.Shared ? null : options.Pool;
+            _pool = options!.Pool == MemoryPool<byte>.Shared ? null : options.Pool;
```

Comment:
This `!` should not be necessary.  You need to put a `[DoesNotReturn]` attribute on all of the `ThrowHelper.ThrowXx` methods that unconditionally throw.